coins/monero/src/wallet/seed/mod.rs

```diff
@@ -0,0 +1,88 @@
+use thiserror::Error;
+
+mod classic;
+
+use classic::{CLASSIC_SEED_LENGTH, CLASSIC_SEED_LENGTH_WITH_CHECKSUM};
+
+/// Error when decoding a seed.
+#[derive(Clone, Copy, PartialEq, Eq, Debug, Error)]
+pub enum SeedError {
+  #[error("malformed seed")]
+  InvalidSeed,
+  #[error("unknown language")]
+  UnknownLanguage,
+  #[error("unexpected number of words in seed")]
+  InvalidSeedLength,
+  #[error("invalid checksum")]
+  InvalidChecksum,
+  #[error("invalid number of words in language word list")]
+  InvalidLanguageWordList,
+  #[error("duplicate prefix is found in the language")]
+  DuplicatePrefix,
+  #[error("prefix is not found in language trimmed words")]
+  UnknownPrefix,
+  #[error("word length size less than language prefix length")]
+  InvalidWordInLanguage,
+}
+
+#[derive(Clone, Copy, PartialEq, Eq, Debug)]
+pub enum LanguageName {
+  English,
+  EnglishOld,
+  German,
+  Spanish,
+  French,
+  Italian,
+  Japanese,
+  Dutch,
+  Portuguese,
+  Russian,
+  ChineseSimplified,
+  Esperanto,
+  Lojban,
+}
+
+// TODO: Add polyseed to enum
+#[derive(PartialEq, Eq, Debug)]
```

Comment:
This should derive Clone and not derive Debug. It should implement Debug, yet not print the String when it does.